sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/models/IdentityPropertiesAutoGenerated.java

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.resourcemanager.containerregistry.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+
+/** Managed identity for the resource. */
+@Fluent
+public final class IdentityPropertiesAutoGenerated {
```

Comment:
This seems a duplicate class. -- `IdentityPropertiesAutoGenerated` and `IdentityProperties`.

We need to avoid this in SDK.

Probably need a directive. @XiaofeiCao for discussion.

Reply:
PS: the tag actually contains a preview swagger. the duplicate may come from it. but since it been there for quite some time, we probably not able to remove that swagger.
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/containerregistry/resource-manager#tag-package-2023-07

Reply:
So should you try using this tag [package-2023-07-only](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/containerregistry/resource-manager#tag-package-2023-07-only) ?

Reply:
We probably cannot, as it was in previous GA versions. Remove that Swagger likely going to cause breaking changes.

Reply:
They changed properties to `readOnly` in `UserIdentityProperties` and added two more properties(`type`, `userAssignedIdentities`) in `IdentityProperties`, hence the additional `AutoGenerated` classes...